rs/index_writer/src/index_writer.rs

```diff
@@ -28,7 +30,7 @@ impl IndexWriter {
     }
 
     // TODO(hicder): Support multiple inputs
-    pub fn process(&mut self, input: &mut impl Input) -> Result<()> {
+    pub fn process(&mut self, input: &mut impl Input) -> Result<(), Box<dyn Error>> {
```

Comment:
why do we have to specify Error here? wouldn't `Result` take care of it? are we using `anyhow::Result`?